alpine container image support

Fusl opened the pull request:
This PR implements support for building a valkey-extension container image using the valkey alpine image as base.

In order to get building for Alpine working, three workarounds had to be implemented:

* Rust/cargo requires `RUSTFLAGS="-C target-feature=-crt-static"` for building on Alpine
* `valkey-search` is currently being cloned using the default branch rather than the versioned tag since the current latest versioned tag does not include [the commit that removes the systemd requirement](https://github.com/valkey-io/valkey-search/commit/eec38de7fb0423fb659b451b79a8e3420299e936) - This workaround should be removed once a new release tag has been cut in the upstream repository.
* The version of the `re2` library required through `grpc` from the `valkey-search` source requires a patch based on [this upstream commit](https://github.com/google/re2/commit/b025c6a3ae05995660e3b882eb3277f4399ced1a) until `grpc` [finalizes their currently work-in-progress upgrade to a newer `re2` version](https://github.com/grpc/grpc/pull/37938)

roshkhatri commented on Dockerfile.template:
```diff
@@ -34,7 +60,14 @@ WORKDIR /opt
+# need to clone main on alpine until this commit is available in a release tag:
+# https://github.com/valkey-io/valkey-search/commit/eec38de7fb0423fb659b451b79a8e3420299e936
+# TODO: update this to use a release tag when available
+    git clone --depth 1 https://github.com/valkey-io/valkey-search.git;
```
This will not be `1.0.0-rc1` version for `valkey-search` for the alpine??

Fusl replied:
I'll pin it to commit https://github.com/valkey-io/valkey-search/commit/eec38de7fb0423fb659b451b79a8e3420299e936 instead which is technically still `1.0.0-rc1` but with the changes to remove the systemd requirement applied.

roshkhatri replied:
Yes that is a good idea, can you add?

Fusl replied:
Added that now

roshkhatri commented on Dockerfile.template:
```diff
@@ -46,16 +79,40 @@ RUN set -eux; \
 # Build Search module
 WORKDIR /opt/valkey-search
+{{ if env.variant == "alpine" then ( -}}
+# on alpine, the first build fails due to this issue: https://github.com/grpc/grpc/issues/38438
+# we just replace mutable int32_t with mutable int in .build-release/.deps/grpc/third_party/re2/util/pcre.h
+# based on this upstream change https://github.com/google/re2/commit/b025c6a3ae05995660e3b882eb3277f4399ced1a
+RUN set -eux;       \
+    ./build.sh || true; \
+    sed -i 's|^  mutable int32_t |  mutable int     |' .build-release/.deps/grpc/third_party/re2/util/pcre.h; \
+    ./build.sh;
```
Is this also a work around for a new tag on `valkey-search`?

or we will have to keep this later as well?

Fusl replied:
We'll have to keep this part here as mentioned in my initial PR comment:

> * The version of the `re2` library required through `grpc` from the `valkey-search` source requires a patch based on [this upstream commit](https://github.com/google/re2/commit/b025c6a3ae05995660e3b882eb3277f4399ced1a) until `grpc` [finalizes their currently work-in-progress upgrade to a newer `re2` version](https://github.com/grpc/grpc/pull/37938)

roshkhatri reviewed (commented):
overall LGTM, Some comments about `valkey-search` versioning

roshkhatri reviewed (approved):
LGTM

Fusl commented:
@roshkhatri I just noticed this section in `generate-stackbrew-library.sh`: https://github.com/valkey-io/valkey-extension/blob/88722ae5568792c8751b017907c95cb4a8fe1a4d/generate-stackbrew-library.sh#L77-L104

Will I need to add an `alpine` variant to that too?

roshkhatri commented:
> @roshkhatri I just noticed this section in `generate-stackbrew-library.sh`:
> 
> https://github.com/valkey-io/valkey-extension/blob/88722ae5568792c8751b017907c95cb4a8fe1a4d/generate-stackbrew-library.sh#L77-L104
> 
> Will I need to add an `alpine` variant to that too?

Yes, it should be something like this:
https://github.com/roshkhatri/valkey-extension/blob/broken-alpine/generate-stackbrew-library.sh

Fusl commented:
> Yes, [...]

Alrighty, I've updated that as well now. I've also gone ahead and rebased my branch on top of the current mainline version to clear the merge conflicts and additionally also updated some remaining places where we used `valkey-extension` to now say `valkey-extensions`. Please let me know if I'm still missing something.

roshkhatri commented:
I think this might not work as we are merging all the digests and the tags to push, waiting for the workflows to complete, we might need to update the workflow to upload two different digests for bookworm and alpine.

But yes, we aren't missing anything from alpine point of view so thats a good thing

roshkhatri commented:
Can you rebase this once , I have added the infra to the workflow to have two versions.

It is not required but would be good to run the workflows on the PR once

Fusl commented:
> Can you rebase this once

Done
